ingestion/functions/parsing/common/python/parsing_lib/parsing_lib.py

```diff
@@ -69,11 +69,20 @@ def prepare_cases(cases, upload_id):
 
     TODO: Migrate source_id/source_url to this method.
     """
-    for case in cases:
-        case["caseReference"]["uploadId"] = upload_id
+    for i in range(len(cases)):
+        cases[i]["caseReference"]["uploadId"] = upload_id
+        cases[i] = remove_nested_nones(cases[i])
     return cases
 
 
+def remove_nested_nones(d):
+    if not isinstance(d, (dict, list)):
+        return d
+    if isinstance(d, list):
```

Comment:
after reviewing a few other parsers from Anya and Felix I think it'd be nice to also remove empty strings, can you add a check in here as well for that please?

Reply:
Sure, can do.